src/main/java/seedu/address/ui/PersonListPanel.java

```diff
@@ -71,6 +73,24 @@ public PersonListPanel(ObservableList<Person> masterList) {
             studentListView.refresh();
             volunteerListView.refresh();
         });
+
+        autoScrollOnChange(studentListView, students);
+        autoScrollOnChange(volunteerListView, volunteers);
+    }
+
+    private static void autoScrollOnChange(ListView<Person> view, ObservableList<Person> list) {
+        list.addListener((ListChangeListener<Person>) change -> {
+            boolean isScrollNeeded = false;
+            while (change.next()) {
+                if (change.wasAdded() || change.wasReplaced()) {
+                    isScrollNeeded = true;
+                }
+            }
+            if (isScrollNeeded && !list.isEmpty()) {
+                int last = list.size() - 1;
+                Platform.runLater(() -> view.scrollTo(last));
```

Comment:
The automatic scroll to the bottom when a Person is added works, however now we have the opposite problem where this scrolling is undesired when a Person at the top of the list is edited.

Maybe it's better to change line 91 to the edited row instead of the last row.

```suggestion
                Platform.runLater(() -> view.scrollTo(change.getFrom()));
```